Jellyfin.Api/Controllers/VideosController.cs

```diff
@@ -233,25 +233,28 @@ public ActionResult MergeVersions([FromQuery, Required] string? itemIds)
                     .First();
             }
 
-            var list = primaryVersion.LinkedAlternateVersions.ToList();
+            var alternateVersionsOfPrimary = primaryVersion.LinkedAlternateVersions.ToList();
 
             foreach (var item in items.Where(i => i.Id != primaryVersion.Id))
             {
                 item.SetPrimaryVersionId(primaryVersion.Id.ToString("N", CultureInfo.InvariantCulture));
 
                 item.UpdateToRepository(ItemUpdateType.MetadataEdit, CancellationToken.None);
 
-                list.Add(new LinkedChild
+                if (!alternateVersionsOfPrimary.Any(i => string.Equals(i.Path, item.Path, StringComparison.OrdinalIgnoreCase)))
```

Comment:
Wouldn't it be better to check with `Id`?

Reply:
It would also work, I copy the same method use bellow

Reply:
Ok I tried checking the id and don't ask  me why it doesn't work